### PR TITLE
fix: Traverse symlink folders

### PR DESCRIPTION
**Pull request by joshhunt**
https://github.com/gulpjs/glob-stream/pull/118 introduced a slight regression where symlinked directories would not be traversed during walkdir. This PR checks if dirents are symlinks + and traverses them as appropriate.

Fixes https://github.com/gulpjs/vinyl-fs/issues/349

### CODE REVIEW

File: test/fixtures/symlinks/symlink-dest
Comment (joshhunt) [inline test/fixtures/symlinks/symlink-dest:1]: This file is actually a symlink - when git clones the repo it'll set up the symlink.

I'm not sure if this is the best thing to do or not? Alternatively could set up the symlink in some other way (pre-test or post-install script? `beforeAll()` in tests? 🤷)

File: index.js [hunk 1]
```diff
@@ -74,6 +74,17 @@ function walkdir() {
 
       if (dirent.isDirectory()) {
         queue.push(nextpath);
+      } else if (dirent.isSymbolicLink()) {
+        // If it's a symlink, check if the symlink points to a directory
+        fs.stat(nextpath, function (err, stats) {
+          if (err) {
+            return cb(err);
+          }
```
Comment (joshhunt) [inline index.js:82]: I'm not really sure what to do about the error handling here. This `cb` tells the queue an error occured, but presumably `dirents.forEach(processDirent);` will continue processing, which could potentially error again...

Should it instead just throw the error and hope queue catches and emits it properly?

File: test/fixtures/symlinks/symlink-dest
Comment (phated) [reply test/fixtures/symlinks/symlink-dest:1]: I wasn't sure if git would set this up correctly on Windows, but the tests seem to pass so I think it's good as-is.

File: index.js (hunk 1 above)
Comment (phated) [reply index.js:82]: Thanks for pointing this out! It has me rethinking how things are queued, but I'll make those changes after I land this

**Review by phated — APPROVED** [review]
@joshhunt Thank you so much for this! You are a true hero in digging into the issue and supplying a fix. Please let me know if there is anything I can do to keep you actively involved in our community <3

**Comment by joshhunt** [comment]
Neat - thanks for the quick review and getting this one in.

Do you think we could cut a release of vinyl-fs with this version? I'm happy to PR whatever's needed for that (just version bump?)

**Comment by phated** [comment]
@joshhunt It's actually unnecessary to release a new vinyl-fs, as we use the semver caret (`^`) to indicate [^8.0.0](https://github.com/gulpjs/vinyl-fs/blob/master/package.json#L28) which will pick up this release automatically! 🎉 Thanks again for your help here.

**Comment by sunnysonx** [comment]
On WSL, it introduced a bug with maximum call stack on i18next-scanner package. 

Despite the fact that I do not have symlinks inside my project, it reaches the maximum call stack.
image

I have downgraded glob-stream to 8.0.0 and it resolved the issue.

All our current projects that were using i18next-scanner started to fail if glob-stream was updated to 8.0.1

**Comment by sunnysonx** [comment]
@joshhunt I'm sorry, idk how it happened that I searched across all involved packages, but it seems that I ommited to search in glob-stream. Thank you

**Pull request merged**